test/integration/comprehensive-transformer-test.spec.ts

```diff
@@ -292,6 +292,67 @@ describe('Comprehensive Transformer Test', () => {
           });
         });
       });
+      describe('transformToExpectedUnderlying', () => {
+        const AMOUNT_PER_UNDERLYING = utils.parseEther('1');
+        when('transforming to an expected amount of underlying', () => {
+          let neededDependent: BigNumber;
+          given(async () => {
+            const input = underlying.map((token) => ({ underlying: token.address, amount: AMOUNT_PER_UNDERLYING }));
+            neededDependent = await transformer.calculateNeededToTransformToUnderlying(dependent.address, input);
+            await dependent.connect(signer).approve(transformer.address, neededDependent);
+            await transformer.connect(signer).transformToExpectedUnderlying(dependent.address, input, RECIPIENT);
+          });
+          then('allowance is spent', async () => {
+            expect(await dependent.allowance(signer.address, transformer.address)).to.equal(0);
+          });
+          then('dependent tokens are transferred', async () => {
+            const balance = await dependent.balanceOf(signer.address);
+            expect(balance).to.equal(INITIAL_SIGNER_BALANCE.sub(neededDependent));
+          });
+          then('expected underlying tokens are sent to the recipient', async () => {
+            for (const underlyingToken of underlying) {
+              const recipientBalance = await underlyingToken.balanceOf(RECIPIENT);
```

Comment:
I was wondering how tf did this work with protocol token, remember the `IERC20Like`. Nice.